pegasus/sites.v3/code.org/views/school_info.haml

```diff
@@ -0,0 +1,102 @@
+      %select#school-type.form-control{name: "#{form_name}[school_type]", type: "select", required: false}
+        %option{value: "", selected: true, disabled: true}
+        %option{value: SchoolInfo::SCHOOL_TYPE_CHARTER} Charter
+        %option{value: SchoolInfo::SCHOOL_TYPE_PRIVATE} Private
+        %option{value: SchoolInfo::SCHOOL_TYPE_PUBLIC} Public
+        %option{value: SchoolInfo::SCHOOL_TYPE_OTHER} Other
+
+    .form-group{style: "display: none;"}
+      .question State
+      %select#school-state.form-control{name: "#{form_name}[school_state]", type: "select", required: false}
+        %option{value: "", selected: true, disabled: true}
+        - get_all_states_with_dc.each do |state_abbr, state_name|
+          %option{value: state_abbr}= state_name
+
+    .form-group{style: "display: none;"}
+      .question District
+      #school-district
+
+        -# We'll apply the Selectize UI control to this input.
+        %input#school-district-id.form-control{name: "#{form_name}[school_district_id]", type: "text", required: false}
+
+        %label
+          %input#school-district-other{name: "#{form_name}[school_district_other]", type: "checkbox", value: "true", style: "margin-top: 0px"}
+          %span{style: "font-size: 12px"}
+            Other district not listed above
+
+        .form-group{style: "display: none;"}
+          .question District Name
+          %div.controls
+            %input#school-district-name.form-control{name: "school_info[school_district_name]", type: "text", required: false, style: "width: calc(100% - 15px);"}
+
+    .form-group{style: "display: none;"}
+      .question School
+      #school
+
+        -# We'll apply the Selectize UI control to this input.
+        %input#school-id.form-control{name: "#{form_name}[school_id]", type: "text", required: false}
+
+        %label
+          %input#school-other{name: "#{form_name}[school_other]", type: "checkbox", value: "true", style: "margin-top: 0px"}
+          %span{style: "font-size: 12px"}
+            Other school not listed above
+
+    #no-schools-warning{style: "display: none; margin-bottom: 5px;"}
+      .question No schools of that type were found in this district.
+
+    -# '#school-name' is outside of '#school' so that we can still show it when the school dropdown is hidden.
+    .form-group{style: "display: none;"}
+      .question
+        School Name
+      %div.controls
+        %input#school-name.form-control{name: "school_info[school_name]", type: "text", required: false, style: "width: calc(100% - 15px);"}
+
+    .form-group{style: "display: none;"}
+      .question School’s ZIP code
+      %input#school-zipcode.form-control{name: "#{form_name}[school_zip]", type: "text", required: false}
+
+    .form-group{style: "display: none;"}
+      .question
+        School Address
+      %div.controls
+        %textarea#school-address.form-control{name: "school_info[full_address]", required: false, rows: 4, style: "width: calc(100% - 15px);"}
```

Comment:
My understanding is that this haml is all copied from somewhere else, and perhaps deeply coupled with the JS in schoolInfo.js? Are we setting ourselves up for trouble by having two different instances of the DOM both tightly coupled with a single instance of some JS?

Reply:
At the moment the DOM is pretty much the same and we don't have plans to have it diverge. We needed a copy in Pegasus since Sinatra can't "reach across" to Dashboard to get the haml. In the unlikely event the DOM diverges in the future and needs different JS, we'll fork the JS at that time.

Reply:
I thought @breville had built a way to share haml between dashboard/pegasus?